src/main/java/com/google/collegeplanner/servlets/APIUtil.java

```diff
@@ -0,0 +1,124 @@
+import java.security.NoSuchAlgorithmException;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.conn.ssl.TrustAllStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.apache.http.util.EntityUtils;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+/** Provides an interface for making outside API calls. */
+public class APIUtil {
+  /**
+   * Returns a json object given a URIBuilder.
+   *
+   * @param builder The endpoint that will be requested
+   */
+  public JSONArray requestAPIAndGetJsonArray(URIBuilder builder) {
+    if (builder == null) {
+      return null;
+    }
+
+    HttpGet apiRequest;
+    try {
+      apiRequest = new HttpGet(builder.build());
+    } catch (URISyntaxException e) {
+      return null;
+    }
+
+    String json = requestAPI(apiRequest);
+    JSONParser parser = new JSONParser();
+    JSONArray jsonArray;
+    try {
+      jsonArray = (JSONArray) parser.parse(json);
+    } catch (ParseException e) {
+      return null;
+    }
+    return jsonArray;
+  }
+
+  /**
+   * Returns a json object given a URIBuilder.
+   *
+   * @param builder The endpoint that will be requested
+   */
+  public JSONObject requestAPIAndGetJsonObject(URIBuilder builder) {
+    if (builder == null) {
+      return null;
+    }
+
+    HttpGet apiRequest;
+    try {
+      apiRequest = new HttpGet(builder.build());
+    } catch (URISyntaxException e) {
+      return null;
+    }
+
+    String json = requestAPI(apiRequest);
+    JSONParser parser = new JSONParser();
+    JSONObject jsonObject;
+    try {
+      jsonObject = (JSONObject) parser.parse(json);
+    } catch (ParseException e) {
+      return null;
+    }
+    return jsonObject;
+  }
+
+  /**
+   * Makes a GET request and returns the reponse json.
+
+   * @param apiRequest The HttpGet object that will be executed
+   */
+  private String requestAPI(HttpGet apiRequest) {
+    // Ignore any SSL certificate validations.
+    String json;
+    try (CloseableHttpClient httpClient =
+             HttpClients.custom()
+                 .setSSLContext(new SSLContextBuilder()
+                                    .loadTrustMaterial(null, TrustAllStrategy.INSTANCE)
+                                    .build())
+                 .setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE)
+                 .build();
+         CloseableHttpResponse apiResponse = httpClient.execute(apiRequest)) {
+      HttpEntity entity = apiResponse.getEntity();
+      json = EntityUtils.toString(entity);
```

Comment:
what about this:

```
try {
  // ... 
  if (entity != null && apiResponse... == SC_OK) {
    return EntityUtils.toString(entity);
  }
} catch {
}

return "";
```

Then you can get rid of the `String json;` declaration on line 103, and all your error/failure cases condense down into a single return line.